service/app/controllers/usermanagement/UserCrudController.java

```diff
@@ -0,0 +1,275 @@
+package controllers.usermanagement;
+
+import static org.sunbird.learner.util.Util.isNotNull;
+import static org.sunbird.learner.util.Util.isNull;
+
+import controllers.BaseController;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.commons.collections.MapUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.sunbird.common.exception.ProjectCommonException;
+import org.sunbird.common.models.util.ActorOperations;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.ProjectLogger;
+import org.sunbird.common.models.util.ProjectUtil;
+import org.sunbird.common.models.util.ProjectUtil.EsType;
+import org.sunbird.common.models.util.StringFormatter;
+import org.sunbird.common.request.BaseRequestValidator;
+import org.sunbird.common.request.HeaderParam;
+import org.sunbird.common.request.Request;
+import org.sunbird.common.request.UserRequestValidator;
+import org.sunbird.common.responsecode.ResponseCode;
+import play.libs.F.Promise;
+import play.mvc.Result;
+import util.AuthenticationHelper;
+
+public class UserCrudController extends BaseController {
+
+  /**
+   * This method will do the registration process. registered user data will be store inside
+   * cassandra db.
+   *
+   * @return Promise<Result>
+   */
+  public Promise<Result> createUser() {
+    return handleRequest(
+        ActorOperations.CREATE_USER.getValue(),
+        request().body().asJson(),
+        (req) -> {
+          Request request = (Request) req;
+          UserRequestValidator.validateCreateUser((Request) request);
+          UserRequestValidator.fieldsNotAllowed(
+              Arrays.asList(JsonKey.ORGANISATION_ID), (Request) request);
+          HashMap<String, Object> innerMap = new HashMap<>();
+          innerMap.put(JsonKey.USER, request.getRequest());
+          request.setRequest(innerMap);
+          return null;
+        },
+        null,
+        null,
+        true);
+  }
+
+  /**
+   * This method will do the registration process. registered user data will be store inside
+   * cassandra db.
+   *
+   * @return Promise<Result>
+   */
+  public Promise<Result> createUserV2() {
+
+    return handleRequest(
+        ActorOperations.CREATE_USER.getValue(),
+        request().body().asJson(),
+        (req) -> {
+          Request request = (Request) req;
+          UserRequestValidator.validateCreateUserV2(request);
+          HashMap<String, Object> innerMap = new HashMap<>();
+          innerMap.put(JsonKey.USER, request.getRequest());
+          request.setRequest(innerMap);
+          return null;
+        },
+        JsonKey.VERSION_2,
```

Comment:
It is confusing to pass JsonKey.VERSION_2 and JsonKey.VERSION as they do not correspond to pathId and pathVariable. Rather explicitly set it in the lambda function within the request.